### ISSUE TITLE
[BUG] RiskGate.check_hard_rules: float('nan') max_loss_usd bypasses all loss checks silently

### ISSUE THREAD

**Issue by JackSmack1971** [JackSmack1971/degenexus#14]
## Forensic Evidence Trace

**File:** `src/core/risk_gate.py:79–84`
```python
max_allowed_loss = portfolio_value * self.limits.max_loss_pct_per_trade
if proposal.max_loss_usd > max_allowed_loss:   # ← comparison with NaN always False
    violations.append(
        f"MAX_LOSS_EXCEEDED: ${proposal.max_loss_usd:.2f} > "
        f"${max_allowed_loss:.2f} ..."
    )
```

**File:** `src/models/signals.py` — `TradeProposal.max_loss_usd: float` — no `isfinite` validator.

**Physical arithmetic proof (confirmed in Python 3.14 runtime):**
```python
>>> import math
>>> nan = float('nan')
>>> nan > 200.0
False          # ← check NEVER fires for NaN
>>> nan >= 0
False          # ← no amount of threshold setting protects against NaN
>>> math.isfinite(nan)
False          # ← the correct guard is absent
```

**grep confirms zero NaN/isfinite guards across all core modules:**
```bash
grep -n "nan\|isfinite\|isnan\|math\." src/core/risk_gate.py src/core/execution_gate.py \
     src/core/portfolio.py src/core/slippage.py src/core/trade_lifecycle.py
# → 0 results (only a comment matched in portfolio.py)
```

**grep confirms zero tests exercise NaN inputs:**
```bash
grep -rn "nan\|isfinite" tests/
# → 0 results
```

## State-Delta Assertion (FSV/SoT)

Expected: Any `TradeProposal` with `max_loss_usd` exceeding `portfolio_value × max_loss_pct_per_trade` is rejected by `check_hard_rules` with `MAX_LOSS_EXCEEDED` violation.

Observed: A `TradeProposal` with `max_loss_usd = float('nan')` passes `check_hard_rules` with zero violations. The trade proceeds to execution with undefined loss exposure. The `MAX_LOSS_EXCEEDED` violation is never appended.

Delta: The hard loss gate (Layer 1) is silently disabled for any NaN financial value. A trade with `max_loss_usd = NaN` executes as if loss=0 while carrying undefined real exposure.

## Blast Radius

- Any execution path that allows a NaN to reach `TradeProposal.max_loss_usd` — possible via:
  - LLM fallback returning 0/null parsed to NaN via arithmetic (e.g. `0 / 0`)
  - `QuantAgent._fallback()` returning bare dict with missing field
  - Python `float('nan')` from `float(None)` if optional field is mishandled
- The hard gate is the ONLY deterministic loss guardrail. Its failure here means no financial protection exists for NaN inputs.
- All four additional `>`, `>=`, `/` comparisons in `check_hard_rules` are equally NaN-vulnerable: `POSITION_LIMIT`, `EXPOSURE_LIMIT` checks also pass silently for NaN inputs.

## Reproduction

```python
from src.core.risk_gate import RiskGate
from src.models.signals import TradeProposal
import math

# Construct proposal with NaN max_loss — Pydantic v2 accepts float('nan')
proposal = TradeProposal(
    ...,
    max_loss_usd=float('nan'),
)
gate = RiskGate()
violations = gate.check_hard_rules(proposal, portfolio_value=10000.0,
    open_positions_count=0, total_exposure_usd=0.0)
assert violations == []  # BUG: passes with no violations
```

## Suggested Next Action

Add `math.isfinite()` guard at the top of `check_hard_rules`:
```python
import math

def check_hard_rules(self, proposal, portfolio_value, open_positions_count, total_exposure_usd):
    non_finite = [f for f in [proposal.max_loss_usd, proposal.position_size_shares,
                               proposal.entry_price, proposal.stop_loss, proposal.take_profit]
                  if not math.isfinite(float(f))]
    if non_finite:
        violations.append(f"INVALID_FINANCIAL_VALUE: non-finite inputs detected")
        return violations  # short-circuit — do not evaluate further
    ...
```
Alternatively, add Pydantic `field_validator` with `math.isfinite()` assertion to `TradeProposal`.

---
Executing agent: agent:claude-sonnet-4-6
Session ID: loop-audit-cycle-3-2026-05-25
Commit SHA: b0c4952ec1ae10f3fe8f9f81f7178abd7702f17d
